[islands] refreshed page content for 4.0 (#5461)
withastro/docs: src/content/docs/en/concepts/islands.mdx

import IslandsDiagram from '~/components/IslandsDiagram.astro';

Astro pioneered and popularized a frontend architecture called **Islands.** Islands architecture results in better frontend performance by helping you avoid monolithic JavaScript patterns and stripping all non-essential JavaScript from the page automatically. Developers keep using their favorite UI components and frameworks with Astro and still get these benefits.

## A brief history

The term "component island" was first coined by Etsy's frontend architect [Katie Sylor-Miller](https://twitter.com/ksylor) in 2019. This idea was then expanded on and documented in [this post](https://jasonformat.com/islands-architecture/) by Preact creator Jason Miller on August 11, 2020.

> The general idea of an "Islands" architecture is deceptively simple: render HTML pages on the server, and inject placeholders or slots around highly dynamic regions [...] that can then be "hydrated" on the client into small self-contained widgets, reusing their server-rendered initial HTML.  
> — Jason Miller, Creator of Preact

The technique that this architectural pattern builds on is also known as **partial** or **selective hydration.**

> In contrast, most JavaScript-based web frameworks hydrate & render an entire website as one large JavaScript application (also known as a single-page application, or SPA).  SPAs provide simplicity and power but suffer from page-load performance problems due to heavy client-side JavaScript usage.

Author: ```suggestion
In contrast, most JavaScript-based web frameworks hydrate & render an entire website as one large JavaScript application (also known as a single-page application, or SPA). SPAs provide simplicity and power but suffer from page-load performance problems due to heavy client-side JavaScript usage.
```

  </Fragment>
  <Fragment slot="carouselApp">Image carousel (interactive island)</Fragment>
  <Fragment slot="footer">Footer (static HTML)</Fragment>

>  <Fragment slot="source">Source: [Islands Architecture: Jason Miller](https://jasonformat.com/islands-architecture/)</Fragment>

Author: ```suggestion
  <Fragment slot="source">Source: [Islands Architecture: Jason Miller](https://jasonformat.com/islands-architecture/)</Fragment>
```

>     <Fragment slot="source">Source: [Islands Architecture: Jason Miller](https://jasonformat.com/islands-architecture/)</Fragment>

Author: ```suggestion
 <Fragment slot="source">Source: [Islands Architecture: Jason Miller](https://jasonformat.com/islands-architecture/)</Fragment>
```

  </Fragment>
  <Fragment slot="carouselApp">Image carousel (interactive island)</Fragment>
  <Fragment slot="footer">Footer (static HTML)</Fragment>
    <Fragment slot="source">Source: [Islands Architecture: Jason Miller](https://jasonformat.com/islands-architecture/)</Fragment>
</IslandsDiagram>

An island always runs in isolation from other islands on the page, and multiple islands can exist on a page. Islands can still share state and communicate with each other, even though they run in different component contexts.

This flexibility allows Astro to support multiple supported UI frameworks like [React](https://react.dev/), [Preact](https://preactjs.com/), [Svelte](https://svelte.dev/), [Vue](https://vuejs.org/), and [SolidJS](https://www.solidjs.com/). Because they are independent, you can even mix several frameworks on each page.

:::tip
Although most developers will stick to just one UI framework, Astro supports multiple frameworks in the same project. This allows you to:

- Choose the framework that is best for each component.
- Learn a new framework without needing to start a new project.
- Collaborate with others even when working in different frameworks.

> - Incrementally convert an existing site to another framework with no down time.

Reviewer: Nit: I believe here "down time" should be spelled as one word.
```suggestion
- Incrementally convert an existing site to another framework with no downtime.
```

:::

## Creating an island

By default, Astro will automatically render every UI component to just HTML & CSS, **stripping out all client-side JavaScript automatically.** 

```astro title="src/pages/index.astro"
<MyReactComponent />
```

> This may sound strict, but this behavior is what keeps Astro websites fast-by-default and protects developers from accidentally sending unnecessary or unwanted JavaScript that might slow down their website.

Reviewer: In the Why Astro page, and the old Islands page, we use "Fast by default" (no hyphen), I would change this for consistency.
```suggestion
This may sound strict, but this behavior is what keeps Astro websites fast by default and protects developers from accidentally sending unnecessary or unwanted JavaScript that might slow down their website.
```
